src/Sentry.NLog/SentryTarget.cs

```diff
@@ -0,0 +1,249 @@
+        public string MinimumBreadcrumbLevel
+        {
+            get => Options.MinimumBreadcrumbLevel?.ToString() ?? LogLevel.Off.ToString();
+            set => Options.MinimumBreadcrumbLevel = LogLevel.FromString(value);
+        }
+
+        #endregion Xml property setters
+
+        #region Lifecycle methods
+
+        /// <inheritdoc />
+        protected override void CloseTarget()
+        {
+            _sdkDisposable?.Dispose();
+            base.CloseTarget();
+        }
+
+        /// <inheritdoc />
+        protected override void InitializeTarget()
+        {
+            base.InitializeTarget();
+
+            IncludeEventProperties = Options.SendEventPropertiesAsData;
+
+            if (Options.MarkNLogNamespaceAsExternal)
+            {
+                Options.AddInAppExclude("NLog");
+            }
+
+            // If the sdk is not there, set it on up.
+            if (Options.InitializeSdk && _sdkDisposable == null)
+            {
+                _sdkDisposable = SentrySdk.Init(Options);
+            }
+        }
+
+        /// <inheritdoc />
+        protected override void Write(LogEventInfo logEvent)
+        {
+            if (logEvent?.Message == null)
+            {
+                return;
+            }
+
+            var hub = _hubAccessor();
+
+            if (hub?.IsEnabled != true)
+            {
+                return;
+            }
+
+            var exception = logEvent.Exception;
+            var formatted = Layout.Render(logEvent);
+            var template = logEvent.Message;
+
+            var contextProps = GetAllProperties(logEvent);
+
+            var shouldOnlyLogExceptions = exception == null && Options.IgnoreEventsWithNoException;
+
+            if (logEvent.Level >= Options.MinimumEventLevel && !shouldOnlyLogExceptions)
+            {
+                var evt = new SentryEvent(exception)
+                {
+                    Sdk =
+                    {
+                        Name = Constants.SdkName,
+                        Version = NameAndVersion.Version
+                    },
+                    Message = null,
+                    LogEntry = new LogEntry
+                    {
+                        Formatted = formatted,
+                        Message = template
+                    },
+                    Level = logEvent.Level.ToSentryLevel(),
+                    Release = Options.Release,
+                    Environment = Options.Environment,
+                };
+
+                evt.Sdk.AddPackage(ProtocolPackageName, NameAndVersion.Version);
+
+                // Always apply any manually configured tags
+                evt.SetTags(GetTags(logEvent));
+
+                if (IncludeEventProperties)
+                {
+                    evt.SetExtras(contextProps);
+                }
+
+                if (Options.SendEventPropertiesAsTags)
+                {
+                    evt.SetTags(GetLoggingEventProperties(logEvent).MapKeys(a => a.ToString()));
```

Comment:
Why the MapKeys-logic when GetLoggingEventProperties is already returning keys as string?

Reply:
I may have misnamed that helper function, haha. It actually maps the existing keys to the newly mapped values. I'll rename that.